File: 0001-two-sum/0001-two-sum.ts
function twoSum(nums: number[], target: number): number[] {
    let indexMap = {};
    
    for (let i=0; i < nums.length; i++) {
        
        if (indexMap[target - nums[i]] !== undefined) {
            return [i, indexMap[target - nums[i]]]
        }
        indexMap[nums[i]] = i;        
    }
        

};

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> IndexMap = new Dictionary<int, int>();
        
        int[] result = {0, 0};
        
        for (int i=0; i < nums.Length; i++) {
            
            if (IndexMap.ContainsKey(target - nums[i])) {
                result[0] = IndexMap[target - nums[i]];
                result[1] = i;
                break;
            }
            
            IndexMap[nums[i]] = i;
        }
        
        return result;
    }
}